data/role/jenkins.yaml

```diff
@@ -1,17 +1,20 @@
 ---
 psick::base::linux_classes:
   jenkins: psick::jenkins
-  openssh: psick::openssh
+  jenkins_casc: psick::jenkins::jcasc
   profile_env: profile::profile
 psick::profiles::linux_classes:
   jenkins: ''
 
 psick::jenkins::disable_setup_wizard: true
-psick::jenkins::admin_password: 'aam5oaQu'
+psick::jenkins::admin_password: ENC[PKCS7,MIIBeQYJKoZIhvcNAQcDoIIBajCCAWYCAQAxggEhMIIBHQIBADAFMAACAQEwDQYJKoZIhvcNAQEBBQAEggEATCj81Q+gBL86HsVCimdXdmCKe0RRXeafKGBGi+b8IydtMPeUr9855alCTtcpUbIXNEvgOPL3ybsZ8mWGz9ZGn2zj4ux1077kkoKivvQ5dznQJGWMSeNXBVDH9aW0cACkHc3vvnMrO4lMyqREz7GkjZkZ8LWahAFG2ILjo2LUcHBY+PXyRAiOZXN1EFterGyiaRJ/x7KuXTPWaQgGo+rFH2z7agh+aUziZN3t4Gd/T8TxBIfOprIkehRXSQBZJflj7jrW9PpViBAJL7/MpVPjqxNHO4qWn1d7aNKIW479uj/D/Iu+pRhzcPiqlrgiaaZmsiPbIunpgX8zkrcUpPOfpjA8BgkqhkiG9w0BBwEwHQYJYIZIAWUDBAEqBBCUOns1dg770kVgz3Jm7XNPgBATBR7F1nDiE0oIJgW1NzN0]
 
 profile::profile::scripts: 
   ec2_tags:
     template: 'profile/aws.tags.erb'
+
+psick::jenkins::jcasc::config_template: 'hieradata/jenkins.yaml.erb'
```

Comment:
+1 for having the template directly in the hieradata module.
Just note that in other cases templates were placed in the profile module in the control-repo (https://github.com/graduway/psick/tree/production/site/profile/templates) so it makes sense to standardise the approach.

Also, I'd suggest to have coherent naming for files and dirs in the templates dir with the classes were they are used.
So this could become  :
psick::jenkins::jcasc::config_template: 'hieradata/jenkins/jenkins.yaml.erb'
and the template moved accordingly.
my2c